tests/test_mbed_targets.py

```diff
@@ -11,9 +11,13 @@
     MbedTargets,
     _get_target,
     _target_matches_query,
+    TargetBuildAttributesError,
 )
 from mbed_targets._internal.configuration import DatabaseMode
 
+TEST_DIR = pathlib.Path(__file__).parents[0]
+PATH_TO_TEST_TARGETS_JSON = TEST_DIR.joinpath("test_mock_sources/test_targets.json")
```

Comment:
Having a separate fixture file like this is a bit fragile. Could we not `pyfakefs` and pass the path to file generated in test instead?

Reply:
Yeah that makes sense, Ill give that a go 👍

Reply:
715d1b0